plugins/node/opentelemetry-instrumentation-aws-sdk/src/services/dynamodb.ts

```diff
@@ -61,7 +61,7 @@ export class DynamodbServiceExtension implements ServiceExtension {
     const operation = response.request.commandName;
 
     if (operation === 'BatchGetItem') {
-      if ('ConsumedCapacity' in response.data) {
+      if ('ConsumedCapacity' in response.data && 'map' in response.data.ConsumedCapacity){
```

Comment:
Thanks for reporting it and creating a fix.
I think using the `in` operator to check if a function is found on the object is a bit non-ideomatic.
What do you think about changing it to: `if(Array.isArray(response.data?.ConsumedCapacity))`?
Also - is it possible to add a unit-test for this case?

Reply:
Thanks for your comment @blumamir . I've pushed some changes to address your comment. [667eade](https://github.com/open-telemetry/opentelemetry-js-contrib/pull/899/commits/667eade433b7e9a49a756861b048b6ecbab9deef)